plugins/plugin_utils/acl_popper.py

```diff
@@ -0,0 +1,138 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2023 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+"""
+The remove_keys plugin code
+"""
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+import re
+
+from ansible.errors import AnsibleFilterError
+
+
+def _raise_error(msg):
+    """Raise an error message, prepend with filter name
+    :param msg: The message
+    :type msg: str
+    :raises: AnsibleError
+    """
+    error = "Error when using plugin 'acl_popper': {msg}".format(msg=msg)
+    raise AnsibleFilterError(error)
+
+
+def clear_empty_data(data):
+    if isinstance(data, dict):
+        # for k in list(data.keys()):
+        #     if not data.get(k, {}):
+        #         del data[k]
+        for k, v in data.items():
+            data[k] = clear_empty_data(v)
+    if isinstance(data, list):
+        temp = []
+        for i in data:
+            if i:
+                temp.append(clear_empty_data(i))
+        return temp
+    return data
+
+
+def check_match(ace, match_criteria):
+    if ace.get("sequence") == match_criteria.get("sequence"):
```

Comment:
Should this require all the match criteria to be met not just one?

Reply:
@cidrblock I am still confused should we make it meet all the matching criteria and fail on what basis? 
what do you suggest?